### ISSUE TITLE
Add support for PostgreSQL ltree column type

### ISSUE THREAD

**Issue by noshua** [vladmihalcea/hypersistence-utils#543]
It would be cool if you could provide a UserType for PostgreSQL ltree-Type too.

**Comment by vladmihalcea** [comment]
@noshua It would be even cooler if you provide it.

Looking forward to reviewing your Pull Request. You can use the existing Types as a reference.

**Comment by Halynsky** [comment]
Hello. It would be nice to have it.
@vladmihalcea Would you please find some time to check PR?

**Comment by vladmihalcea** [comment]
@Halynsky The current PR is only targeting the 6.0 branch, so it would take me way quite some time to port it to the other modules.

And, since I haven't ever used the `ltree` column type, I'd have to first learn about it to see if the implementation is correct.

All in all, that's the equivalent of one day of consulting. Does your company want to sponsor this time I need to invest in integrating this issue?

**Comment by Halynsky** [comment]
@vladmihalcea 
Sorry. It's not a commercial project. It's helping Ukraine.
But I will consider such kind of help if something will be required for some other commercial project. Thx.

I have used ltree in two projects. Had my own old implementation.
Now we migrated to Spring 3.0.1 and Hibernate 6.1.6
hypersistence-utils-hibernate-60 (3.2.0)

So I had to rework the implementation. And found this issue and PR.
Just for your information. I used the source code of this PR and it works for me.
